apps/condo/domains/resident/tasks/helpers/sendBillingReceiptsAddedNotifications.js

```diff
@@ -1,226 +1,56 @@
-
-                // NOTE: (DOMA-4351) skip already notified user to get rid of duplicate notifications
-                if (notifiedUsers.has(resident.user.id)) continue
-
-                const success = await prepareAndSendNotification(context, receipt, resident)
-
-                if (success) notifiedUsers.add(resident.user.id)
-
-                successConsumerCount += success
-            }
-
-            if (successConsumerCount > 0) successCount += 1
-
-            lastReceipt = receipt
-        }
-
-        // Store receipt.createdAt as lastDt in order to continue from this point on next execution
-        if (isFunction(onLastDtChange) && !isEmpty(lastReceipt)) await onLastDtChange(lastReceipt.createdAt)
-
-        logger.info({ msg: `Processed ${skip} receipts of ${receiptsCount}.` })
-    }
-    logger.info({ msg: 'sent billing receipts', successCount, receiptsCount })
-}
-
-
-const sendBillingReceiptsAddedNotifications = async (resendFromDt = null) => {
-    const { prevMonthStart, thisMonthStart } = getStartDates()
+const sendBillingReceiptsAddedNotifications = async () => {
     const redisClient = getRedisClient()
-    const handleLastDtChange = async (createdAt) => { await redisClient.set(REDIS_LAST_DATE_KEY, createdAt) }
+    const { prevMonthStart } = getStartDates()
+    const lastSync = new Date(prevMonthStart).toISOString()
 
     /**
      * This represents min value for billingReceipt createdAt to start processing from
      * 1. First of all use resendFromDt from CLI parameter if available, else
      * 2. Use createdAt value from last success script execution stored in Redis, else
      * 3. Use thisMonthStart
      */
-    const lastDt = resendFromDt ? resendFromDt.replace(' ', 'T') : await redisClient.get(REDIS_LAST_DATE_KEY) || thisMonthStart
 
-    logger.info({ msg: 'stored date', storedDt: await redisClient.get(REDIS_LAST_DATE_KEY) })
+    logger.info({ msg: 'Starting billing receipts notification process' })
+
+    // After synchronization with integrations (RegisterBillingReceiptsService), the context is updated to put the lastReport:JSON field.
+    // Therefore, we understand that according to the context updated after the date of the last notification mailing, there are new receipts, and we fetch such contexts.
 
-    const receiptsWhere = {
-        period_in: [prevMonthStart, thisMonthStart],
-        createdAt_gt: lastDt,
+    const BillingContexts = await find('BillingIntegrationOrganizationContext', {
+        updatedAt_gte: lastSync,
+        status: CONTEXT_FINISHED_STATUS,
         deletedAt: null,
+    })
+    const oldWayLastSyncDate = await redisClient.get(REDIS_LAST_DATE_KEY)
+
+    for (const context of BillingContexts) {
+        const lastReport = get(context, 'lastReport.finishTime')
+        if (!lastReport) continue
+        logger.info({ msg: 'Old way redis key', oldWayLastSyncDate, lastReport, result: dayjs(oldWayLastSyncDate).isAfter(dayjs(lastReport)) })
```

Comment:
Please move uncommon fields (`oldWayLastSyncDate `, `lastReport` and `result`) to `data` to prevent new indexes in logs database